test/blackbox-tests/test-cases/private-package-lib/main.t

```diff
@@ -87,7 +87,6 @@ We want to see mangled names in the dune-package file as well:
     (byte __private__/secret/secret.cma)
     (native __private__/secret/secret.cmxs))
    (native_archives __private__/secret/secret.a)
-     (source (path Secret) (impl (path __private__/secret/secret.ml))))))
```

Comment:
Where did this entry go?

Reply:
this is a `grep` for `private`. because the path was removed, it no longer appears in the filter.

Reply:
Yeah, but shouldn't it remain inside __private__? Otherwise, where do we keep the sources for private libraries attached to a package?

Reply:
I feel like this is correct. There's no `__private__` folder in the source tree. 

The way I understand it is that `secret.ml` is a sibling of the `dune` file that defines its library, so its path should reflect that.

Reply:
> The way I understand it is that secret.ml is a sibling of the dune file that defines its library, so its path should reflect that.

I agree, the dune file defines the library's root in the source. In the installed layout, that root should be where the library archives are. In this case, they're in `__private__/$lib/`

Where is `secret.ml` after this change then? Doesn't that mean there's a possibility that two different libraries with the same source files can conflict while they're in the same package. I'm not even sure if we have a test for that.

Reply:
I added a `dune install` call to this test in [91cbede](https://github.com/ocaml/dune/pull/7153/commits/91cbedee26d8ed714b1c6e523252b1bcf3ed9fb5). 

It looks like `secret.ml` is still installed the correct location. My understanding is that the `(path ..` entry under sources in the dune-package maps it to the location starting from the library root. this is now correct, as the module `secret` is next to the `dune` file.

Let me know if I'm still missing something.

Reply:
> I added a dune install call to this test in [91cbede](https://github.com/ocaml/dune/pull/7153/commits/91cbedee26d8ed714b1c6e523252b1bcf3ed9fb5).

Yes, these install locations look good.

> It looks like secret.ml is still installed to the correct location. My understanding is that the (path .. entry under sources in the dune-package maps it to the location starting from the library root. this is now correct, as the module secret is next to the dune file.

All other paths in the `dune-package` are interpreted relative to the `dune-package` file. See the archive file right above for example.

Reply:
check https://github.com/ocaml/dune/pull/7153/commits/82434c20c42b5ea1a79df759481dbd857b65f98d